src/main/java/org/apache/jmeter/protocol/mqtt/paho/clients/BlockingClient.java

```diff
@@ -64,24 +64,21 @@ public BlockingClient(String brokerUrl, String clientId, boolean cleanSession, S
         // stored until the message has been delivered to the server.
         //..a real application ought to store them somewhere
         // where they are not likely to get deleted or tampered with
-        String testPlanFileDir = System.getProperty("java.io.tmpdir");
-
-        if (null != GuiPackage.getInstance()) {
-            String testPlanFile = GuiPackage.getInstance().getTestPlanFile();
-            testPlanFileDir = FilenameUtils.getFullPathNoEndSeparator(testPlanFile);
-            testPlanFileDir = testPlanFileDir + File.separator + "tmp" + File.separator + clientId + File.separator + Thread.currentThread().getId();
-        }
+        //String tmpDir = System.getProperty("java.io.tmpdir");
 
+        String testPlanFile = GuiPackage.getInstance().getTestPlanFile();
```

Comment:
Thanks that will be a problem for me too.  We can use
FileServer.getFileServer().getScriptName(); instead